src/structures/CommandInteraction.js

```diff
@@ -0,0 +1,149 @@
+  }
+
+  /**
+   * Defers the reply to this interaction.
+   * @param {boolean} [ephemeral] Whether the reply should be ephemeral
+   */
+  async defer(ephemeral) {
+    await this.client.api.interactions(this.id, this.token).callback.post({
+      data: {
+        type: InteractionResponseTypes.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+        data: {
+          flags: ephemeral ? MessageFlags.FLAGS.EPHEMERAL : undefined,
+        },
+      },
+    });
+  }
+
+  /**
+   * Options for a reply to an interaction.
+   * @extends MessageOptions
+   * @property {boolean} [ephemeral] Whether the reply should be ephemeral
+   * @typedef {Object} InteractionReplyOptions
+   */
+
+  /**
+   * Creates a reply to this interaction.
+   * @param {StringResolvable|APIMessage} content The content for the reply
+   * @param {InteractionReplyOptions|MessageAdditions} [options] Additional options for the reply
+   */
+  async reply(content, options) {
+    const apiMessage = content instanceof APIMessage ? content : APIMessage.create(this, content, options);
+    const { data } = await apiMessage.resolveData().resolveFiles();
+
+    await this.client.api.interactions(this.id, this.token).callback.post({
+      data: {
+        type: InteractionResponseTypes.CHANNEL_MESSAGE_WITH_SOURCE,
+        data,
+      },
+    });
+  }
+
+  /**
+   * Represents an option of a received command interaction.
+   * @property {string} name The name of the option
+   * @property {ApplicationCommandOptionType} type The type of the option
+   * @property {string|number|boolean} [value] The value of the option
+   * @property {CommandInteractionOption[]} [options] Additional options if this option is a subcommand (group)
+   * @property {User} [user] The resolved user
+   * @property {?GuildMember} [member] The resolved member
+   * @property {?GuildChannel} [channel] The resolved channel
+   * @property {?Role} [role] The resolved role
+   * @typedef {Object} CommandInteractionOption
+   */
+
+  /**
+   * Transforms an option received from the API.
+   * @param {Object} option The received option
+   * @param {Object} resolved The resolved interaction data
+   * @returns {CommandInteractionOption}
+   * @private
+   */
+  transformOption(option, resolved) {
+    const result = {
+      name: option.name,
+      type: ApplicationCommandOptionTypes[option.type],
+    };
+
+    if (option.value) {
+      result.value = option.value;
+    }
+
+    if (option.options) {
+      result.options = option.options.map(o => this.transformOption(o, resolved));
+    }
+
+    if (option.type === ApplicationCommandOptionTypes.USER) {
+      const user = resolved.users[option.value];
+      result.user = this.client.users.add(user);
+
+      const member = resolved.members?.[option.value];
+      if (member) {
+        member.user = user;
+        result.member = this.guild?.members.add(member) ?? null;
```

Comment:
is there a need to use optional chaining here? a member would only be provided if there is a guild

Reply:
This is a fail-safe in case the guild is not cached